spec/mailers/request_mailer_spec.rb

```diff
@@ -48,12 +48,29 @@
       expect(InfoRequest.holding_pen_request.incoming_messages.count).to eq(1)
     end
 
-    it "puts messages with the request address in Bcc: in the holding pen" do
-      request = FactoryBot.create(:info_request)
-      receive_incoming_mail('bcc-contact-reply.email',
-                            email_to: 'dummy@localhost',
-                            email_bcc: request.incoming_email)
-      expect(InfoRequest.holding_pen_request.incoming_messages.count).to eq(1)
+
+    it "attaches messages with an info request address in the Received headers to the appropriate request" do
```

Comment:
This is a good test to keep even when the main implementation is moved to `MailHandler` but we should add additional test for the changes made there.